pkg/controller/template/render.go

```diff
@@ -80,6 +83,11 @@ func generateTemplateMachineConfigs(config *RenderConfig, templateDir string) ([
 			continue
 		}
 
+		// Avoid creating resources for non arbiter deployments
+		if role == arbiterRole && config.Infra.Status.ControlPlaneTopology != configv1.HighlyAvailableArbiterMode {
```

Comment:
nit: technically the controlplanetopology type

https://github.com/openshift/api/blob/52377f09e5c9293d4694c29375b35fb202bae222/config/v1/types_infrastructure.go#L94-L103

isn't a required field. In practice does this always exist? There's a few checks for this in the code, just want to make sure that this can't panic

Reply:
That's a good point, in practice that should always exist, the installer will create a default value for that infra resource as well, but you're right in that the specific instance render config infra object is a pointer, I'll add some extra checks to make sure it can't panic, but it doesn't seem like we check in other instances so i'll add a helper function to avoid that